### ISSUE TITLE
We should have a button to copy the MXID from the profile sidebar

### ISSUE THREAD

**Issue by dkasak** [element-hq/element-web#26690]
#### What would you like to do?

Have an easy way of copying the user's MXID.

#### Why would you like to do it?

Because I want to copy the MXID in order to use it (e.g. to send it to someone).

#### How would you like to achieve it?

By having a copy button here:

![image](https://github.com/vector-im/element-web/assets/463342/b62d9af2-1405-4a6f-9da4-3af9c0637cd2)

### Have you considered any alternatives?

There's some duplication with `Share Link to User` functionality. In practice, I think one very rarely wants to share their MXID on Facebook/Reddit/Twitter/etc directly from EW/ED. The QR code could be a bit more useful if we had similar functionality for actually adding users by QR code. All in all, I think having a quick and easy way of copying a MXID is much more useful than  `Share Link to User`.

Potentially we want to copy the matrix.to URL instead. However, I think https://github.com/vector-im/element-web/issues/21867 should just be fixed to enable conversion of MXIDs to matrix.to URLs and pillification.

### Additional context

This is similar to https://github.com/vector-im/element-web/issues/10204 but for the profile page instead of the top left menu. A copy button there would be nice too though, to enable easy sharing of your own MXID with other people.

**Comment by t3chguy** [comment]
There is already a "Share Link to User" button a few lines down

**Comment by dkasak** [comment]
> There is already a "Share Link to User" button a few lines down

Listed in alternatives, but deemed suboptimal.

**Comment by t3chguy** [comment]
Its worth keeping in mind that pasting/writing an MXID doesn't cause a mention in an intentional mentions world, only pills do

**Comment by dkasak** [comment]
You're right. But in that case, the fact that links to a user pillify (and therefore mention them) is horrible, given links are what is currently emitted by the `Share Link to User` functionality, which to me implies simply sharing a reference to a user and not mentioning them.

In any case, I agree this whole area needs design. To be honest, I think it's very surprising links to users and MXIDs are treated differently since there is a 1-to-1 correspondence between them.

I also think there's some conflation over the meaning of pills too; they seem to serve both as "pretty UI rendering of a reference to a Matrix resource" and "thing that triggers a mention". For rooms/spaces, we use the first meaning solely. But for users, we add the latter into the mix, which is where the problem arises.

So maybe we need two kinds of pills ("just a reference" pill vs an... intentionally mentioning pill).

**Comment by gaelledel** [comment]
This may be addressed with the work on the room header plus right panel @americanrefugee - Can you check if we're happy to close this issue?

**Comment by americanrefugee** [comment]
Yes, I'm happy to close this issue. "Share profile" includes the ability to simply copy the MXID.

**Comment by dkasak** [comment]
Is there a place to check this new design, to see whether it actually addresses my issue?

Having to click into `Share profile` sounds like it introduces another extra click to the operation (which is quite common in some workflows, such as security), not to mention it introduces quite a bit of cognitive burden because "Share profile" doesn't quite align with "Copy the user's ID" at least in my mind.

Is there any specific reason not to introduce a quicker, more accessible way to do this, other than it being extra work?

**Comment by americanrefugee** [comment]
Ok, I've added a copy button next to the MXID: https://www.figma.com/design/rTaQE2nIUSLav4Tg3nozq7/Compound-Web-Components?node-id=1121-78832&t=ExGCIBE0jnTovBQ1-4

**Comment by TemoCrevlis** [comment]
This issue has been fixed (finished)?
